src/resolve/resolved-dns-packet.c

```diff
@@ -2174,6 +2174,8 @@ int dns_packet_extract(DnsPacket *p) {
                                 if (r < 0)
                                         goto finish;
                         }
+
+                        previous = rr;
```

Comment:
Yupp this is not obvious. Also the code is actually borked, thanks for making me think about it! 

So the thinking here is that if we decided to keep the RR around (either in `p->opt`, because it's the EDNS0 OPT RR, or in the `answer` object), then we remember that, so that if the next RR we read from the packet shares the same RR key, we can merge the RR keys into one, and reduce memory usage a bit. But as it turns out I fucked this up: dns_answer_add() will only add an RR to the answer if it doesn't exist in the answer yet, otherwise it is a NOP. Now, since previous didn't actually take a reference of its own, this meant that if somebody sends us a packet with the same identical RR listed twice, then previous will point to an unreferenced RR on the next iteration, and boom!

Hence, thanks for making me think about this. Will fix this so that we take an explicit reference here, and add a comment what this code does!